src/js/menus.layer-context-menu.dropchop.js

```diff
@@ -0,0 +1,89 @@
+var dropchop = (function(dc) {
+
+  'use strict';
+
+  dc = dc || {};
+  dc.menus = dc.menus || {};
+  dc.menus.layerContextMenu = dc.menus.layerContextMenu || {};
+
+  dc.menus.layerContextMenu.init = function() {
+    // build ops.geo container
+    var geoContainer = $('<div>').addClass('operations-geo');
+    dc.$elem.append(geoContainer);
+
+    // Bind the context-menu handler on any item added to layerlist
+    $(dc).on('layerlist:added', function(e, layer){
+      $(layer).bind("contextmenu", function(e) {
+        // Clear other open menus
+        dc.menus.layerContextMenu.removeLayerContextMenus();
+
+        // Select right-clicked layer.
+        // TODO: In future, support context-menu on many layers
+        $(e.target).trigger('click');
+
+        // create left-click menu
+        var menu = $('<div>').addClass('context-menu');
+        menu.css({top: e.pageY, left: e.offsetX});
+        var menuList = $('<ul>');
+        menu.append(menuList);
+
+        var menuItems = dc.menus.layerContextMenu._getMenuOperations();
+        for (var menuItem in menuItems) {
+          var fileBtn = $('<li>').addClass('menu-action')
+            .html(menuItems[menuItem].icon + menuItems[menuItem].description)
+            .attr('data-operation', menuItem)
+            .attr('data-tooltip', menuItems[menuItem].description);
+            // .text(menuItems[menuItem].description);
+            if (menuItems[menuItem].type === 'info') fileBtn.addClass('dropchop-info');
+          fileBtn.on('click', _handleBtnClick);
+          menuList.append(fileBtn);
+        }
+        $(this).parent().append(menu);
+
+        e.preventDefault();
+      });
+    });
+
+    // Remove open menus
+    $('html').bind('click', function (e) {
+      dc.menus.layerContextMenu.removeLayerContextMenus();
+    });
+  };
+
+  // Remove any open context-menus from DOM
+  dc.menus.layerContextMenu.removeLayerContextMenus = function() {
+    $('.context-menu').remove();
+  };
+
+  dc.menus.layerContextMenu._getMenuOperations = function() {
+    return {
+      /* jshint ignore:start */
+      'extent': dc.ops.file['extent'],
+      'expand': dc.ops.file['expand'],
+      'combine': dc.ops.file['combine'],
+      'rename': dc.ops.file['rename'],
+      'remove': dc.ops.file['remove'],
```

Comment:
This was written before I saw how #208 rewrote how functions were loaded with the Left Sidebar Menu.  This could probably be refactored to more closely match that style for the sake of simplicity.  One win that this style has is that it could allow operations to be pulled from different files.